utils/exporting/score_encoding_xls.py

```diff
@@ -41,14 +43,87 @@ def export_xls(cohort):
     return save_virtual_workbook(workbook)
 
 
+def export_xls_with_scores(cohort, periods, students, internships):
+    workbook = Workbook()
+    worksheet = workbook.active
+    _add_header(cohort, worksheet)
+    _make_complete_list(periods, students, worksheet)
+    _make_internship_sheets(internships, periods, students, workbook)
+    return save_virtual_workbook(workbook)
+
+
+def _make_internship_sheets(internships, periods, students, workbook):
+    for internship in sorted(internships):
+        workbook.create_sheet(internship)
+        worksheet = workbook[internship]
+        _add_sheet_header(worksheet)
+        _add_sheet_content(internship, periods, students, worksheet)
+
+
+def _add_sheet_content(internship, periods, students, worksheet):
+    for student in students:
+        columns = []
+        columns.append(student.person.last_name.upper())
+        columns.append(student.person.first_name)
+        columns.append(student.registration_id)
+        _complete_student_row_by_internship(columns, internship, periods, student)
+        add_row(worksheet, columns)
+
+
+def _complete_student_row_by_internship(columns, internship, periods, student):
+    for period in periods:
+        if period.name in student.specialties.keys() and student.specialties[period.name] == internship:
+            _append_row_data(columns, period, student)
+
+
+def _append_row_data(columns, period, student):
+    if period.name in student.organizations.keys():
+        columns.append(student.organizations[period.name])
+    if period.name in student.periods_scores.keys():
+        columns.append(student.periods_scores[period.name])
+    else:
```

Comment:
CC : Similar blocks of code found in 2 locations. Consider refactoring.
utils/exporting/score_encoding_xls.py:105